File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        vector<int> res={-1,-1};
        int f=0;
        int l=nums.size()-1;

        while(f<=l){
            int mid=f+(l-f)/2;
            if(nums[mid]==target){
                res[0]=mid;
                l=mid-1;
            }
            else if(nums[mid]<target){
                f=mid+1;

            }
            else{
                l=mid-1;
            }
        }
        f=0;
        l=nums.size()-1;
        while(f<=l){
            int mid=(l+f)/2;
            if(target==nums[mid]){
                res[1]=mid;
                f=mid+1;
            }
            else if(nums[mid]<target){
                f=mid+1;
            }
            else{
                l=mid-1;
            }
        }
return res;

    
        
    }
};

File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public static int searchIdx(int[] arr, int tar,boolean fi){
        int s=0,e=arr.length-1;
        int ans=-1;
        while(s<=e){
            int mid=s+(e-s)/2;
            if(arr[mid]>tar){
                e=mid-1;
            }
            else if(arr[mid]<tar){
                s=mid+1;
            }
            else{
                ans=mid;
                if(fi){
                    e=mid-1;
                }
                else{
                    s=mid+1;
                }
            }
        }
        return ans;
    }
    public int[] searchRange(int[] nums, int target) {
        int[] ans={-1,-1};
        int fi=searchIdx(nums,target,true);

        int li=searchIdx(nums,target,false);
        ans[0]=fi;
        ans[1]=li;
        return ans;
    }
}